### DISCUSSION TITLE
How to set containerd proxy after installation?

### DISCUSSION THREAD

**Discussion by Magicloud** [k3s-io/k3s#11316] [Q&A] [answered]
I am using NixOS, hence I did not install K3S via the script, but Nixpkgs. And due to I am using ZFS, I used external Containerd, instead of the one from K3S.

Look through the [doc](https://docs.k3s.io/advanced#configuring-an-http-proxy), `CONTAINER_HTTPS_PROXY` is set in an environment file. So I made the service file as:

```
# /etc/systemd/system/k3s.service
[Unit]
After=firewall.service network-online.target
Description=k3s service
Wants=firewall.service network-online.target

[Service]
Environment="LOCALE_ARCHIVE=/nix/store/m4n7lnvkrdnymlpa8pwy1pwnr93s1vgh-glibc-locales-2.40-36/lib/locale/locale-archi>
Environment="PATH=/nix/store/qhkb7c916jj34rfgics7rss2424hg4gb-zfs-user-2.2.6/bin:/nix/store/sf6y4arqcm100rnnl3dhpg732>
Environment="TZDIR=/nix/store/5z7yfmp96ijpx0lhy5gpj65i38972330-tzdata-2024b/share/zoneinfo"
Delegate=yes
EnvironmentFile=/etc/k3s.service.env
ExecStart=/nix/store/gfy7ivfgns7c5cijq8dycfsa0y6a2w40-k3s-1.31.1+k3s1/bin/k3s server \
 --container-runtime-endpoint unix:///run/containerd/containerd.sock --write-kubeconfig-mode 644
ExecStartPre=/nix/store/28b6fzix8dqpqss4i6wipnjydp27yd4a-activate-k3s-content
KillMode=process
LimitCORE=infinity
LimitNOFILE=1048576
LimitNPROC=infinity
Restart=always
RestartSec=5s
TasksMax=infinity
Type=notify

[Install]
WantedBy=multi-user.target
```
And in /etc/k3s.service.env
```
CONTAINERD_HTTPS_PROXY=http://192.168.0.102:8080/
```

It did not seem working. I also tried directly put the env var in Service.Environment of systemd. No luck.

Where should I set the proxy? Or with external Containerd, I should set the proxy there, not K3S?

**Comment by dereknola** [comment, marked as answer]
Your last hunch was correct. If you are running external Containerd, you configure it following their[ standard docs](https://github.com/containerd/containerd/blob/main/docs/PLUGINS.md). All the K3s docs are about configuring the embedded containerd that we ship with.

**Reply by Magicloud** [reply]
Thank you.
